src/lib/Persistence/Legacy/Content/Mapper/ResolveVirtualFieldSubscriber.php

```diff
@@ -0,0 +1,215 @@
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Core\Persistence\Legacy\Content\Mapper;
+
+use eZ\Publish\Core\FieldType\NullStorage;
+use eZ\Publish\Core\Persistence\Legacy\Content\FieldValue\Converter\Exception\NotFound;
+use eZ\Publish\Core\Persistence\Legacy\Content\FieldValue\ConverterRegistry;
+use eZ\Publish\Core\Persistence\Legacy\Content\Gateway as ContentGateway;
+use eZ\Publish\Core\Persistence\Legacy\Content\StorageFieldValue;
+use eZ\Publish\Core\Persistence\Legacy\Content\StorageRegistry;
+use eZ\Publish\SPI\Persistence\Content\Field;
+use eZ\Publish\SPI\Persistence\Content\FieldValue;
+use eZ\Publish\SPI\Persistence\Content\Type\FieldDefinition;
+use eZ\Publish\SPI\Persistence\Content\VersionInfo;
+use Ibexa\Contracts\Core\Event\Mapper\ResolveMissingFieldEvent;
+use Ibexa\Contracts\FieldType\DefaultDataFieldStorage;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+final class ResolveVirtualFieldSubscriber implements EventSubscriberInterface
+{
+    /** @var \eZ\Publish\Core\Persistence\Legacy\Content\FieldValue\ConverterRegistry */
+    private $converterRegistry;
+
+    /** @var \eZ\Publish\Core\Persistence\Legacy\Content\StorageRegistry */
+    private $storageRegistry;
+
+    /** @var \eZ\Publish\Core\Persistence\Legacy\Content\Gateway */
+    private $contentGateway;
+
+    public function __construct(
+        ConverterRegistry $converterRegistry,
+        StorageRegistry $storageRegistry,
+        ContentGateway $contentGateway
+    ) {
+        $this->converterRegistry = $converterRegistry;
+        $this->storageRegistry = $storageRegistry;
+        $this->contentGateway = $contentGateway;
+    }
+
+    public static function getSubscribedEvents(): array
+    {
+        return [
+            ResolveMissingFieldEvent::class => [
+                ['persistExternalStorageField', -100],
+                ['resolveVirtualExternalStorageField', -80],
+                ['resolveVirtualField', 0],
+            ],
+        ];
+    }
+
+    public function resolveVirtualField(ResolveMissingFieldEvent $event): void
+    {
+        if ($event->getField()) {
+            return;
+        }
+
+        $content = $event->getContent();
+
+        try {
+            $emptyField = $this->createEmptyField(
+                $content->versionInfo,
+                $event->getFieldDefinition(),
+                $event->getLanguageCode()
+            );
+
+            $event->setField($emptyField);
+        } catch (NotFound $exception) {
+            return;
+        }
+    }
+
+    /**
+     * @throws \eZ\Publish\Core\Persistence\Legacy\Content\FieldValue\Converter\Exception\NotFound
+     */
+    public function persistExternalStorageField(ResolveMissingFieldEvent $event): void
+    {
+        $field = $event->getField();
+
+        if ($field && $field->id !== null) {
```

Comment:
$field is not of a boolean type
```suggestion
        if ($field !== null && $field->id !== null) {
```